compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -1243,9 +1243,12 @@ pub fn fn_can_unwind(tcx: TyCtxt<'_>, fn_def_id: Option<DefId>, abi: SpecAbi) ->
         | RiscvInterruptM
         | RiscvInterruptS
         | CCmseNonSecureCall
-        | Wasm
         | PlatformIntrinsic
         | Unadjusted => false,
+        Wasm => {
+            tcx.sess.panic_strategy() == PanicStrategy::Unwind
+                || tcx.sess.target_features.contains(&Symbol::intern("exception-handling"))
+        }
```

Comment:
Unsure about this; I'm pretty sure the behavior is correct but it might not be necessary yet.

Reply:
`extern "wasm"` shouldn't need exception support.

Reply:
Oh - just to be sure, this is for whether it's possible for an `extern "wasm"` function to unwind, for whether catchpads should be generated or not, right? Like, the wasm abi allows for unwinding (when `+exception-handling`), but maybe that's not what `extern "wasm"` is supposed to mirror?

Reply:
Oh, gotcha.